azure-pipelines.yml

```
@@ -23,17 +23,50 @@ jobs:
   steps:
   - script: |
       printenv
-      echo hello
       pwd
       ls
+      echo "agend build directory"
+      echo $(Agent.BuildDirectory)
     displayName: filesystem check
   - script: |
+      cd $(Agent.BuildDirectory)/s
       git clone https://github.com/oci-labs/rosidl_typesupport_opendds.git
     displayName: clone rosidl_typesupport_opendds
   - script: |
+        cd $(Agent.BuildDirectory)
+        git clone https://github.com/adamsj-oci/rmw_build
+        vcs import s < rmw_build/ros2.repos
+        echo "ls"
+        ls
+        echo "ls s"
+        ls s/
+    displayName:  rmw build repo and dependencies
+  - script: |
+        cd $(Agent.BuildDirectory)
+        source /opt/OpenDDS/setenv.sh
+        source /opt/ros/eloquent/setup.bash
+        printenv|grep DDS
+        printenv|grep ROS
+        colcon build --symlink-install --packages-up-to rmw_opendds_cpp
+    displayName:  build up to rmw_opendds_cpp
+  - script: |
+        cd $(Agent.BuildDirectory)
+        source /opt/OpenDDS/setenv.sh
+        source /opt/ros/eloquent/setup.bash
+        source $(Agent.BuildDirectory)/install/local_setup.bash
+        printenv|grep DDS
+        printenv|grep ROS
+        colcon build --symlink-install --packages-up-to rcl_interfaces examples_rclcpp_minimal_publisher examples_rclcpp_minimal_subscriber
+    displayName:  build simple tests
+  - script: |
+        cd $(Agent.BuildDirectory)
         source /opt/OpenDDS/setenv.sh
         source /opt/ros/eloquent/setup.bash
+        source $(Agent.BuildDirectory)/install/local_setup.bash
```

```suggestion
        source install/local_setup.bash
```
Not necessarily wrong... (also on 56)

---

The $(Agent.BuildDirectory) is required because of the layout of the workspace. Repos are cloned into $(Agent.BuildDirectory)/s and the build scripts `rmw_build` are cloned to $(Agent.BuildDirectory). The build scripts expect these relative paths.

---

My point is that it's already your working directory